internal/clients/http/azure/azure_service_client.go

```diff
@@ -0,0 +1,147 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+)
+
+type serviceClient struct {
+	credential *azidentity.ClientSecretCredential
+}
+
+func init() {
+	clients.GetServiceAzureClient = newServiceClient
+}
+
+func newServiceClient(ctx context.Context) (clients.AzureService, error) {
+	opts := azidentity.ClientSecretCredentialOptions{}
+	identityClient, err := azidentity.NewClientSecretCredential(config.Azure.TenantID, config.Azure.ClientID, config.Azure.ClientSecret, &opts)
+	if err != nil {
+		return nil, fmt.Errorf("unable to init Azure credentials: %w", err)
+	}
+
+	return &serviceClient{
+		credential: identityClient,
+	}, nil
+}
+
+func (c *serviceClient) RegisterInstanceTypes(ctx context.Context, instanceTypes *clients.RegisteredInstanceTypes, regionalTypes *clients.RegionalTypeAvailability) error {
+	restricted := make(map[armcompute.ResourceSKURestrictionsReasonCode]int, 0)
+
+	skuClient, err := armcompute.NewResourceSKUsClient(config.Azure.SubscriptionID, c.credential, nil)
+	if err != nil {
+		return fmt.Errorf("unable to generate types: %w", err)
+	}
+	skuPager := skuClient.NewListPager(nil)
+	for skuPager.More() {
+		nextResult, pageErr := skuPager.NextPage(ctx)
+		if pageErr != nil {
+			return fmt.Errorf("unable to generate types: %w", pageErr)
+		}
+		for _, v := range nextResult.Value {
+			if *v.ResourceType != "virtualMachines" {
+				continue
+			}
+			instanceType := clients.InstanceType{
+				Name:        clients.InstanceTypeName(*v.Name),
+				AzureDetail: &clients.InstanceTypeDetailAzure{},
+			}
+			vcpusPerCore := int32(0)
+			for _, c := range v.Capabilities {
+				if *c.Name == "CpuArchitectureType" {
+					instanceType.Architecture, pageErr = clients.MapArchitectures(ctx, *c.Value)
+					if pageErr != nil {
+						return fmt.Errorf("unable to generate types: %w", pageErr)
+					}
+				}
+				if *c.Name == "vCPUs" {
```

Comment:
I expect the *c.Name to have only one of these values, should we maybe use `else if` just for the better readability?

Reply:
Maybe a switch case of c.Name?

Reply:
Switch it is.